src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -114,5 +121,28 @@ private static ulong ParseUInt64(IHeadersCollection headers, string headerName)
 
             return default;
         }
+
+        private static string ParseString(IHeadersCollection headers, string headerName)
+        {
+            var headerValues = headers.GetValues(headerName).ToList();
```

Comment:
Given that the double enumeration would only occur in a presumably rare scenario (in the nominal case, the header should either be present and correct, or completely absent), I think it's worth removing the .ToList and have a double enumeration when needing to log: https://github.com/DataDog/dd-trace-dotnet/pull/779/files#diff-4ed8d83e5894486f094857b43cd46cd2R87

Reply:
Actually I don't think we should log at all. An empty origin header isn't invalid per-se, maybe we should treat it as a missing header

Reply:
Ok, I'll remove the `.ToList()` and the `Count()` to avoid double enumeration, that means that I'll be removing also the log.